hoot-core/src/main/cpp/hoot/core/io/DataConverter.cpp

```diff
@@ -351,6 +351,8 @@ void DataConverter::_fillElementCache(const QString& inputUrl, ElementCachePtr c
   while (streamReader->hasMoreElements())
   {
     ElementPtr pNewElement = streamReader->readNextElement();
+    if (!pNewElement)
+      continue;
```

Comment:
Elements with the `action="delete"` attribute from JOSM come back as NULL from `readNextElement()` this skips the following NULL pointer usage.